src/core_modules/capture-core/utils/validators/form/isValidNonFutureDate.js

```diff
@@ -3,6 +3,10 @@ import i18n from '@dhis2/d2-i18n';
 import moment from 'moment';
 import { parseDate } from '../../converters/date';
 
+const CUSTOM_VALIDATION_MESSAGES = {
+    INVALID_DATE_MORE_THAN_MAX: i18n.t('A date in the future is not allowed'),
+};
+
 export const isValidNonFutureDate = (value: string) => {
```

Comment:
The future date error message is not displayed when selecting an enrollment date in the future, instead the "Please provide a valid date" error is displayed. Can this be fixed? 
I also think `isValidNonFutureDate` should receive the `internalComponentError` object from the [validators](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/D2Form/field/validators/getValidators.js#L121):
`validator: (value: string, allowFutureDate, internalComponentError) => (allowFutureDate ? true : isValidNonFutureDate(value, internalComponentError)),`

Reply:
Yes, we should just check whether the internal error code from the component is the MAX_DATE one and return our own custom capture app error message if that is the case.